api/segmentation/class-segmentation-client-data.php

```diff
@@ -72,66 +68,7 @@ public function report_client_data( $request ) {
-			$mailchimp_api_key_option_name_legacy = 'newspack_newsletters_mailchimp_api_key';
-			global $wpdb;
-			$mailchimp_api_key = $wpdb->get_row( // phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
-				$wpdb->prepare( "SELECT option_value FROM `$wpdb->options` WHERE option_name IN (%s,%s) ORDER BY FIELD(%s,%s)", $mailchimp_api_key_option_name, $mailchimp_api_key_option_name_legacy, $mailchimp_api_key_option_name, $mailchimp_api_key_option_name_legacy )
-			);
-			if ( $mailchimp_api_key ) {
-				$mc            = new Mailchimp( $mailchimp_api_key->option_value );
-				$campaign_data = $mc->get( "campaigns/$mailchimp_campaign_id" );
-				if ( isset( $campaign_data['recipients'], $campaign_data['recipients']['list_id'] ) ) {
-					$list_id = $campaign_data['recipients']['list_id'];
-					$members = $mc->get( "/lists/$list_id/members", [ 'unique_email_id' => $mailchimp_subscriber_id ] )['members'];
-
-					if ( ! empty( $members ) ) {
-						$subscriber                                  = $members[0];
-						$client_data_update['email_subscriptions'][] = [
-							'email' => $subscriber['email_address'],
-						];
-
-						if ( ! isset( $subscriber['merge_fields'] ) ) {
-							return;
-						}
-
-						$donor_merge_field_option_name      = 'newspack_popups_mc_donor_merge_field';
-						$donor_merge_fields                 = $wpdb->get_row( // phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
-							$wpdb->prepare( "SELECT option_value FROM `$wpdb->options` WHERE option_name = %s LIMIT 1", $donor_merge_field_option_name )
-						);
-						$donor_merge_fields                 = isset( $donor_merge_fields->option_value ) ? explode( ',', $donor_merge_fields->option_value ) : [ 'DONAT' ];
-						$has_donated_according_to_mailchimp = array_reduce(
-							// Get all merge fields whose name contains one of the Donor Merge Field option strings.
-							array_filter(
-								array_keys( $subscriber['merge_fields'] ),
-								function ( $merge_field ) use ( $donor_merge_fields ) {
-									$matches = false;
-									foreach ( $donor_merge_fields as $donor_merge_field ) {
-										if ( strpos( $merge_field, trim( $donor_merge_field ) ) !== false ) {
-											$matches = true;
-										}
-									}
-									return $matches;
-								}
-							),
-							// If any of these fields is "true", the subscriber has donated.
-							function ( $result, $donation_merge_field_name ) use ( $subscriber ) {
-								if ( 'true' === $subscriber['merge_fields'][ $donation_merge_field_name ] ) {
-									$result = true;
-								}
-								return $result;
-							},
-							false
-						);
-
-						if ( $has_donated_according_to_mailchimp ) {
-							$client_data_update['donations'][] = [
-								'mailchimp_has_donated' => true,
-							];
-						}
-					}
-				}
-			}
+			$this->report_mailchimp( $client_id, $mailchimp_campaign_id, $mailchimp_subscriber_id );
```

Comment:
Isn't this redundant now? If the MC API is queried when our API decides whether to display the prompts, why should it be queried again when the API is getting client data report?

Reply:
I think you're right. My initial thought was that the subscriber/donor status may change if the user takes action on that first page, but we capture that activity directly in those cases (if the reader subscribes or donates via a form, for example). 893864e removes the second call on client data report.